wgpu-core/src/command/render.rs

```diff
@@ -1891,7 +1897,7 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
                         };
                         state.is_ready(indexed).map_pass_err(scope)?;
 
-                        let last_vertex = first_vertex + vertex_count;
+                        let last_vertex = first_vertex.saturating_add(vertex_count);
```

Comment:
More concretely, using saturating_add will result in the error message being incorrect - I feel like using `u64` for these intermediate calculations (and the vertex_limit) would be a lot cleaner.

Reply:
People can end up in these seemingly insane error cases when they're learning, doing the math all wrong, leading to passing crazy things to us - letting the user correlate the error message to a value in their code is very important in that case.